.github/GSOC.md

```diff
@@ -0,0 +1,123 @@
+   what you are interested in working on.
+5. **Create a Draft Proposal.** Once you have a good understanding of the project and have an idea
+   of what you would like to work on, create a draft proposal. We are happy to help you refine your
+   proposal and provide feedback. The best way to share your draft proposal is through Google docs.
+6. **Submit Your Proposal.** Now that you have refined your proposal, submit it through the
+   GSoC website **before the final submission deadline**.
+
+## :star: Additional Tips
+
+- **Start Early.** The earlier you start, the more time you will have to get to know the project,
+    accumulate contributions, and refine your proposal.
+- **Communicate Effectively.** We are looking for participants who can communicate effectively. This
+    includes using clear and concise language, being responsive to feedback, and being proactive.
+- **Contribute, Contribute, Contribute.** This is the single best way to show us that you are
+    serious becoming a valuable member of our community.
+
+## :technologist: Your Proposal
+
+Your proposal is your opportunity to show us that you are the right person for the job. To this end,
+we have provided a list of things that we would like to see in your proposal. We are not looking for
+a long proposal, but we are looking for a well-thought-out proposal that shows us that you have
+taken the time to understand the project and have a clear plan for how you will contribute.
+
+### Proposal Outline
+
+#### Introduction
+
+All projects aim to solve a problem or improve something. Start your proposal by introducing the
+problem you are trying to solve or the improvement you are trying to make. This is your opportunity
+to show us that you understand the problem, why it is important, and why it is worth solving.
+Include any relevant background information that will help us understand the problem, like
+statistics, examples, references to existing issues, etc.
+
+#### Project Goals
+
+This should be concise and to the point. What are you trying to accomplish? What are the
+deliverables? What are the success criteria? What are the key milestones? Effective goals are
+specific, measurable, and achievable. This section may be a bulleted list or outline.
+
+#### Implementation Plan
+
+Describe your solution to the problem. This should include a high-level overview of the
+architecture, a list of tasks, and a list of potential challenges, as well as some technical
+details. We want to see that you have thought through the problem and can visualize how you will
+solve it.
+
+#### Timeline
+
+Show us that you can break down your project into manageable tasks and that you have a clear plan
+for how you will accomplish your goals. This should be a timeline that shows what you will be
+working on each week. It should include milestones and deliverables. An effective way to present
+this is in a table or a Gantt chart.
+
+#### About You
```

Comment:
We usually ask GSOC participants what days and times they are available for communication/work so we can plan collaboration days, answer questions faster, or get an idea of when progress will be made.